tests/arc/tcursorloop.nim

```diff
@@ -0,0 +1,45 @@
+discard """
+  cmd: '''nim c --gc:arc --expandArc:traverse --hint:Performance:off $file'''
+  nimout: '''
+--expandArc: traverse
+
+var
+  it
+  jt_cursor
+try:
+  `=copy`(it, root)
+  block :tmp:
+    while (
+      not (it == nil)):
+      if true:
+        echo [it.s]
+        `=copy`(it, it.ri)
+  jt_cursor = root
+  if (
+    not (jt_cursor == nil)):
+    echo [jt_cursor.s]
+    jt_cursor = jt_cursor.ri
+finally:
+  `=destroy`(it)
+-- end of expandArc ------------------------
```

Comment:
@arnetheduck Here is the testcase

Without this change `jt` is not optimzied as `jt_cursor`

```nim
--expandArc: traverse

var
  it
  jt
try:
  `=copy`(it, root)
  block :tmp:
    while (
      not (it == nil)):
      if true:
        echo [it.s]
        `=copy`(it, it.ri)
  `=copy`(jt, root)
  if (
    not (jt == nil)):
    echo [jt.s]
    `=copy`(jt, jt.ri)
finally:
  `=destroy`(jt)
  `=destroy`(it)
-- end of expandArc ------------------------
```